### DISCUSSION TITLE
3D Ansicht in Navigation / 3D view in navigation

### DISCUSSION THREAD

**Discussion by ToToGSARider** [devemux86/cruiser#693] [Q&A] [answered]
Beitrag für Stand der Dinge und Aktualisierung in Cruiser 5 / Contribution to the state of affairs and update in Cruiser 5

**Comment by ToToGSARider** [comment]
[ToToGSARider](https://github.com/ToToGSARider)
[7 hours ago](https://github.com/devemux86/cruiser/discussions/690#discussioncomment-11614248)
Good morning, there has been huge progress in the Cruiser again. How are you getting on with reintroducing 3D and tilting the navigation? Have you already found a solution to the problem? Would it be possible to open a separate article for this topic that you can follow so that you don't always have to write in a general article?

Guten Morgen, es gibt ja wieder riesige Fortschritte im Cruiser. Wie kommt ihr den voran das Thema 3D bzw. das kippen der Navigation wieder einzubauen. Hat man schon Lösungsansätze für das Problem gefunden? Könnte man evtl. für das Thema einen separaten Artikel aufmachen dem man folgen kann, damit man nicht immer in einen allgemeinen Artikel schreiben muss?

@[devemux86](https://github.com/devemux86)
devemux86
[7 hours ago](https://github.com/devemux86/cruiser/discussions/690#discussioncomment-11614307)
Maintainer
Author
No, there is no such plan, as it belongs to [Mapsforge](https://github.com/mapsforge/mapsforge) map library.
If someone implements it, then I can include it in Cruiser as well.

@[ToToGSARider](https://github.com/ToToGSARider)
ToToGSARider
[6 hours ago](https://github.com/devemux86/cruiser/discussions/690#discussioncomment-11614623)
Can't you make a request for Mapsforge? I'm not as technically familiar with it as you are, as I'm only a user. I wouldn't even know where to write, let alone how to describe it.
What does the OsmAnd software use for map material? Is 3D fully available and have I ever read about hillshading in the app?
As I said, I'm just a user. As already suggested, I don't want to keep asking in the general posts, but would find a separate post useful, which you can then simply keep an eye on. I know that it's not so great for you either if I keep asking in the posts.

Könnt ihr nicht eine Anfrage nach Mapsforge stellen? Ich kenne mich da technisch nicht so aus wie Ihr, da ich nun einmal nur Anwender bin. Ich wüßte nicht einmal wo ich hinschreiben muß und schon gar nicht, wie ich das beschreiben soll.
Was nutzt den die Software OsmAnd für Kartenmaterial? Da ist 3D komplett verfügbar und Hillshading habe ich da auch schon einmal in der App gelesen?
Wie gesagt, ich bin nur Anwender. Wie auch schon vorgeschlagen, möchte ich nicht ständig in den allgemeinen Beiträgen nachfragen, sondern fände einen getrennten Beitrag sinnvoll, den man dann einfach im Auge behalten kann. Ich weiss, dass es für euch ja auch nicht so toll ist, wenn ich immer wieder in die Beiträge Frage.

@[devemux86](https://github.com/devemux86)
devemux86
[6 hours ago](https://github.com/devemux86/cruiser/discussions/690#discussioncomment-11614712)
Maintainer
Author
I am the main developer of the [Mapsforge](https://github.com/mapsforge/mapsforge) map library.
It's an open source library, everyone can write and contribute new features. 🙂
Like the new https://github.com/devemux86/cruiser/discussions/612#discussioncomment-11607589.

You can create and monitor a new thread here to request a new feature.

**Comment by ToToGSARider** [comment]
@devemux86 

What I don't quite understand is, if you are the developer of Mapsforge, why can't you put 3D back in?

was ich jetzt nicht ganz erstehe, wenn du der Entwickler von Mapsforge bist, warum kannst du dann 3D nicht wieder einbauen?

**Comment by devemux86** [comment]
@ToToGSARider Thanks for the suggestion.

It depends on [Mapsforge](https://github.com/mapsforge/mapsforge) library that currently does not have 3D view.
If someone implements it, then we can include it in Cruiser as well.

**Comment by DolomitesTom** [comment]
Android 15 seems to be a pipe-dream anyway, causing massive problems to the point of rendering a smartphone unusable. Other manufacturers will probably skip the update and go straight for Android 16. Maybe Open GL will work flawlessly there again. By then, perhaps someone will have fixed the specific problems of Open GL. I've never had any problems with it myself.
How flexible is Cruiser when it comes to replacing the map engine with Open GL again in future?

**Comment by ToToGSARider** [comment]
Couldn't you get in touch with the programmer of OsmAnd here in Github? OsmAnd says something about Open GL in the settings. In return Cruiser could support the programmers of OsmAnd with an import of Kurviger files. Then both companies would have closed gaps in their system again. Just a thought from a simple user.

Könnte man nicht mit dem Programmierer von OsmAnd in Kontakt treten Hier in Github? Bei OsmAnd ist irgend etwas von Open GL zu lesen bei den Einstellungen. Im Gegenzug könnte doch Cruiser die Programmierer von OsmAnd unterstützen mit einem Import von Kurviger Dateien. Dann hätten beide Unternehmen wieder Lücken geschlossen in Ihrem System. Nur mal so ein Gedanke von einem einfachen Anwender.

![Screenshot_20241220_062144_OsmAnd+.jpg](https://github.com/user-attachments/assets/c11b3481-1872-4601-92d2-5090dcdc1d38)

**Reply by 0709wiwiwi** [reply]
I am not sure but I think osmand uses opengl which was largely maintained by devemux86.  Osmand is apparently just not in such a hurry to stop offering opengl already. I think the osmand dev's might be able to contribute here then as well.
https://github.com/mapsforge/vtm/issues/1125

**Comment by devemux86** [comment]
Vulkan will replace OpenGL on Android. From 2026, new devices will only support OpenGL through a translation layer.

If you search the issues of map apps that use OpenGL, you will see that they all have rendering problems.
This is why many users still prefer version 1 of the OsmAnd map rendering engine.

OsmAnd all these years is an amazing app with map rendering engine version 1.

Locus until recently didn't have map rotation until I implemented it in Mapsforge:
- https://github.com/mapsforge/mapsforge/issues/7#issuecomment-2147721066

- https://www.locusmap.app/new-locus-map-4-26-dynamic-offline-lomap-rotation-and-other-features-not-only-for-cyclists/

**Comment by ToToGSARider** [comment]
I have the displayed setting on my Samsung S24+ and have had a look at the view on my phone and in Android Auto. I bought OsmAnd back then in a promotion before I bought the K1, as they didn't have 3D back then, only 2D. 3D was the reason for me to go for K1 and then K2 and then Cruiser. 
As I said, I'm just a user and have no idea about programming and all the technical terms that are used here.

Ich habe die angezeigte Einstellung auf meinem Samsung S24+ und habe die Ansicht am Handy und in Android Auto mal angeschaut. Ich habe mir OsmAnd damals in einer Aktion gekauft, bevor ich mir die K1 gekauft habe, da sie damals kein 3D sondern nur 2D hatten. 3D war ja für mich der Grund auf K1 bzw. dann auch K2 und dann auf Cruiser zu gehen. 
Wie gesagt, bin nur Anwender und habe von programmieren und den ganzen Fachbegriffen die hier so genutzt werden keine Ahnung.

**Comment by ToToGSARider** [comment]
A very simple question: 3D was possible in Cruiser 4. 
Couldn't you just leave it up to the user, as with OsmAnd, to decide whether he wants to use the map material as in Cruiser 4 and the high contrast with 3D or the new map material without 3D and therefore nicely colored? Then theoretically all the problems would be solved, wouldn't they?

Mal eine ganz einfache Frage: in Cruiser 4 war ja 3D möglich. 
Könnte man nicht wie bei OsmAnd dem Anwender einfach die Entscheidung überlassen ob er das Kartenmaterial nutzen möchte wie in Cruiser 4 und dem hohen Kontrast mit 3D oder dann eben das neue Kartenmaterial ohne 3D und dafür schön bunt? Dann sind doch theoretisch alle Probleme behoben, oder ?

**Comment by DolomitesTom** [comment]
I think that's a very good idea. Why not run on two tracks? Android 16 has already been announced for June next year. Perhaps OpenGL will run there without any problems. Many people don't use their main phone for motorcycle navigation anyway, which won't make the leap to Android 15. Under Android 14 or lower, you don't have any problems with OpenGL at all. Would it be a lot of effort to introduce such an option to let the user choose which technology they want to use?

**Comment by ToToGSARider** [comment]
Theoretically, a current cell phone, like my Samsung S24+, should not have any problems with the technical requirements under Android 15. As I said, OsmAnd runs completely without problems with 3D. I really think you should give it a try.

Theoretisch sollte ja ein aktuelles Handy, wie mein Samsung S24+, ja auch keine Probleme bekommen mit den technischen Voraussetzungen unter Android 15. Wie gesagt, OsmAnd läuft bei mir komplett ohne Probleme mit 3D. Ich denke wirklich, dass man es echt probieren sollte.

**Comment by ToToGSARider** [comment]
Good morning everyone, is there any progress on Cruiser 5 yet? I keep reading that something is happening with the new design. However, as there are so many conversations in the conversations, it's easy to lose track here.


Guten Morgen zusammen, gibt es hier jetzt schon erste Fortschritte für Cruiser 5? Ich lese immer wieder, dass bei dem neuen Design was passiert. Da es aber so viele Unterhaltungen in den Unterhaltungen gibt, verliert man hier einfach den Überblick.

**Reply by devemux86** [reply]
@ToToGSARider It is still the same answer, see [above](https://github.com/devemux86/cruiser/discussions/693#discussioncomment-11618761).

**Reply by ToToGSARider** [reply]
@devemux86 

Okay, hope dies last. I'll install Cruiser 5 on the replacement phone and keep checking for updates. Both our subscriptions are canceled at the end of June and until then we'll still be using Cruiser 4 on the main phones.

Okay, die Hoffnung stirbt zuletzt. Ich werde mir auf dem Ersatz Handy Cruiser 5 installieren und nach den Updates dann immer wieder schauen. Unsere beiden Abos sind ja zu Ende Juni gekündigt und bis dahin nutzen wir auf den Haupthandys noch Cruiser 4.

**Reply by MotoUKRider** [reply]
@ToToGSARider I have also been using v4 on my spare phone with the current (but slightly modified) Mapsforge theme. It works very nice and you get to enjoy the improved layout+rendering of objects-vs-zoom, but in 3D.  I also get back the correct GPS Altitude (my S9 requires the now removed MSL correction feature).

It also enables me to use a non-essential device on the bike - and leaves my main phone free to take pictures and have a more comfortable ride in my pocket :roller_coaster: :motorcycle: 

Here's an example:
![cruiser-v4-demo](https://github.com/user-attachments/assets/460de128-a772-489f-88f9-ab2e8d871150)

Of course, this is fine if you are more interested in 3D and are happy with the feature-set (and any bugs!) of v4.xx so you will miss new enhancements/fixes that v5.xx has to offer, like the advanced Hillshading and routing algorithms.

If you wish to use the current map theme in your older v4 Cruiser, place this theme ZIP into `\Android\data\gr.talent.cruiser\files\maps\_themes` and load it from the theme picker within the App (or place it anywhere on your phone and load it externally if you have the extended options enabled).
[MotoRiderVTM.zip](https://github.com/user-attachments/files/18632063/MotoRiderVTM.zip)

This is completely independent of the App and offered "as is".

**Comment by ToToGSARider** [comment]
@MotoUKRider 

We (my wife and I) plan everything completely with Kurviger and only use the Cruiser to drive the Kurviger routes and always drive with recalculation deactivated. We were particularly looking forward to the adaptation for the WunderLINQ in Cruiser 5. The map material of Cruiser 4 with full contrast and 3D was perfect for us and that's why we subscribed to Cruiser twice. We don't need the algorithms for the calculation. The perfect solution for us would be to use the old map material from Cruiser 4 with the new functions for the WunderLINQ in Cruiser 5.

Wir (meine Frau und ich) planen alles komplettt mit kurviger und nutzen den Cruiser nur um die Kurviger Routen zu fahren und fahren auch immer mit deaktivierter Neuberechnung. Wir hatten uns insbesondere bei Cruiser 5 auf die Anpassung für das WunderLINQ gefreut. Das Kartenmaterial vopn Cruiser 4 mit vollem Kontrast und 3D war für uns perfekt und deshalb haben wir Cruiser dann auch 2-mal aboniert. Die Algorythmen für die Berechnung benötigen wir nicht. Für uns wäre die perfekte Lösung, dass alte Kartnematerial aus Cruiser 4 mit den neuen Funktionen für das WunderLINQ in Cruiser 5.

**Comment by devemux86** [comment]
@ToToGSARider 

The 3D view in the navigation will be available again in the next Cruiser.

Die 3D-Ansicht in der Navigation wird im nächsten Cruiser wieder verfügbar sein.

- https://github.com/devemux86/cruiser/discussions/726

<img src="https://github.com/user-attachments/assets/01ffb8ed-e8c2-4d7c-bfce-b3e52288f3e5" width="50%"/>

**Reply by ToToGSARider** [reply]
Yeah, ich Danke Dir. Die Sektflasche wird sofort geöffnet und es geht ein riesiges Dankeschön an euch aus Deutschland. Abo wird dann direkt wieder auf beiden Geräten aktiviert.
Yeah, thank you. The champagne bottle is opened immediately and a huge thank you goes out to you from Germany. Subscription will then be reactivated directly on both devices.

**Comment by ToToGSARider** [comment]
Welche Version hat dann das Update bzw. ist es dann im Playstore verfügbar? Ich habe aktuell noch Cruiser 4 installiert. 
What version does the update have and is it available in the Playstore? I currently still have Cruiser 4 installed.

**Reply by devemux86** [reply]
@ToToGSARider 

We will have a Beta period first and then it will be on Google Play. There will be announcements soon.

Wir werden zuerst eine Beta-Periode haben und dann wird es auf Google Play sein. Es wird bald Ankündigungen geben.

**Comment by ToToGSARider** [comment]
What is the probability that we will have a version by Sunday that Cruiser 5 can already be used for the follow-up mode?


Wie hoch ist die Wahrscheinlichkeit, dass wir bis Sonntag eine Version haben,  dass man Cruiser 5 schon für den Folgemodus nutzen kann?

**Reply by devemux86** [reply]
@ToToGSARider 

We'll see, when it's ready! :slightly_smiling_face: 

Wir werden sehen, wenn es fertig ist! :slightly_smiling_face:

**Reply by DolomitesTom** [reply]
Too bad, I would also be ready to test it for the current Sunday tour :-)

**Reply by ToToGSARider** [reply]
I caught something at the motorcycle show in Dortmund and am lying flat. That's it for riding at the weekend, I won't be able to do it.

Ich habe mir auf der Motorradmesse in Dortmund etwas eingefangen und liege flach. Das war es mit dem fahren am Wochenende, da wird bei mir nichts draus. 😭

**Comment by DolomitesTom** [comment]
Any chance that a test version with VTM is available for the next day trip on the weekend?

**Reply by devemux86** [reply]
@DolomitesTom It's possible, I'm doing some final tests. :slightly_smiling_face:

**Comment by devemux86** [comment]
3D view in navigation is again available in:
- https://github.com/devemux86/cruiser/discussions/734

With VTM (OpenGL) map rendering engine.

**Reply by ToToGSARider** [reply]
That looks SUPER. Will the long press and select 3D function be available under the compass again? 

Das sieht SUPER aus. Wird es auch wieder unter dem Kompass die Funktion geben, lange drücken und 3D auswählen?

**Reply by devemux86** [reply]
@ToToGSARider 

We can long press the compass and depending on the active mode we see:
- In "Navigation" and "Follow location" modes: the "Map orientation" options with 3D
- In "Browse map" mode: the "Location orientation" options for the location arrow

Wir können den Kompass lange drücken und je nach aktivem Modus sehen wir:
- In den Modi "Navigation" und "Standort verfolgen": die Optionen "Kartenausrichtung" mit 3D
- Im Modus "Karte durchsuchen": die Optionen für die "Standortausrichtung" für den Standortpfeil

**Comment by devemux86** [comment, marked as answer]
- 3D view in navigation is again available in https://github.com/devemux86/cruiser/discussions/757

With VTM (OpenGL) map rendering engine.
